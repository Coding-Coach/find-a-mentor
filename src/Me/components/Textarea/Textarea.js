import React, { useContext } from 'react';
import styled from 'styled-components';
import { formFieldContext } from '../FormField/formContext';

const StyledTextarea = styled.textarea`
  font-family: Lato, sans-serif;
  font-size: 14px;
  line-height: 17px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #bfbfbf;
  padding: 7px 12px 6px 8px;
  color: #4f4f4f;
  ::placeholder {
    color: #898889;
  }
  height: 75px;
`;

export const Textarea = props => {
  const id = useContext(formFieldContext);
  return <StyledTextarea id={id} {...props} />;
};
