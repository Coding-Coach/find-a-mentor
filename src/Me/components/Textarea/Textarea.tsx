import React, { TextareaHTMLAttributes, useContext } from 'react';
import styled from 'styled-components/macro';
import { formFieldContext } from '../FormField/formContext';

export const StyledTextarea = styled.textarea<{ invalid?: boolean }>`
  font-family: Lato, sans-serif;
  font-size: 14px;
  line-height: 17px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid
    ${props => (props.invalid ? 'var(--form-text-invalid)' : '#bfbfbf')};
  padding: 7px 12px 6px 8px;
  color: #4f4f4f;
  min-height: 75px;

  ::placeholder {
    color: ${props =>
      props.invalid
        ? 'var(--form-text-invalid)'
        : 'var(--form-text-placeholder)'};
  }

  :disabled {
    opacity: 0.7;
  }
`;

export const Textarea = (
  props: TextareaHTMLAttributes<HTMLTextAreaElement>
) => {
  const id = useContext(formFieldContext);
  return <StyledTextarea id={id} {...props} />;
};
