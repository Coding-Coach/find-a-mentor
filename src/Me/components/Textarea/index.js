export { Textarea as default } from './Textarea';
