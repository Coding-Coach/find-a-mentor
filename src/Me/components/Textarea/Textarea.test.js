import React from 'react';
import Textarea from './';
import { render } from '@testing-library/react';

describe('Textarea component', () => {
  it('renders properly', () => {
    const { container } = render(<Textarea placeholder="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
