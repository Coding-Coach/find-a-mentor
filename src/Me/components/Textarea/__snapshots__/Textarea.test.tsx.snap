// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component renders properly 1`] = `
.c0 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  line-height: 17px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #bfbfbf;
  padding: 7px 12px 6px 8px;
  color: #4f4f4f;
  min-height: 75px;
}

.c0::-webkit-input-placeholder {
  color: var(--form-text-placeholder);
}

.c0::-moz-placeholder {
  color: var(--form-text-placeholder);
}

.c0:-ms-input-placeholder {
  color: var(--form-text-placeholder);
}

.c0::placeholder {
  color: var(--form-text-placeholder);
}

<textarea
  class="c0"
  placeholder="test"
/>
`;
